packages/image/src/extractColors.ts

```diff
@@ -0,0 +1,76 @@
+interface PaletteOptions {
+  colorCount?: number;
+  quality?: number;
+}
+
+/**
+ * Extracts a palette of colors from a given image URL based on provided options.
+ * @param {string} imageUrl - The URL of the image.
+ * @param {PaletteOptions} [options] - The extraction options.
+ */
+export const extractColors = (imageUrl: string, options?: PaletteOptions) =>
+  loadImageFromUrl(imageUrl)
+    .then(extractImage)
+    .then((data) => {
+      if (!data) {
+        throw new Error(
+          "Failed to extract pixel data from the image. Ensure the image URL is valid and accessible.",
+        );
+      }
+
+      const pixels = filterRelevantPixels(data, options?.quality);
+      const count = options?.colorCount || pixels.length;
+      return pixels.slice(0, count);
+    });
+
+/**
+ * Extracts pixel data from an image.
+ * @param {HTMLImageElement} img - The image element.
+ */
+const extractImage = (img: HTMLImageElement) => {
+  const canvas = document.createElement("canvas");
+  const context = canvas.getContext("2d");
+
+  if (!context) {
+    return null;
+  }
+
+  canvas.width = img.naturalWidth;
+  canvas.height = img.naturalHeight;
+  context.drawImage(img, 0, 0);
+
+  return context.getImageData(0, 0, canvas.width, canvas.height).data;
+};
```

Comment:
```suggestion
/**
 * Extracts a palette of colors from a given image URL based on provided options.
 * @param {string} imageUrl - The URL of the image.
 * @param {PaletteOptions} [options] - The extraction options.
 */
export const extractColors = (imageUrl: string, options?: PaletteOptions) =>
  loadImageFromUrl(imageUrl)
    .then(extractPixelDatas)
    .then((pixelDatas) => {
      if (!pixelDatas) {
        throw new Error(
          "Failed to extractPixelDatas from the image. Ensure the image URL is valid and accessible.",
        );
      }

      const pixels = filterRelevantPixels(pixelDatas, options?.quality);
      const count = options?.colorCount || pixels.length;
      return pixels.slice(0, count);
    });

/**
 * Extracts pixel datas from an image.
 * @param {HTMLImageElement} img - The image element.
 */
const extractPixelDatas = (img: HTMLImageElement) => {
  const canvas = document.createElement("canvas");
  const context = canvas.getContext("2d");

  if (!context) {
    return null;
  }

  canvas.width = img.naturalWidth;
  canvas.height = img.naturalHeight;
  context.drawImage(img, 0, 0);

  return context.getImageData(0, 0, canvas.width, canvas.height).data;
};
```

Reply:
How about `extractPixels`? When naming it specifically, the term `data` doesn't seem to add much meaningful context